backend/eventd/entity_test.go

```diff
@@ -24,116 +29,237 @@ func TestCreateProxyEntity(t *testing.T) {
 		wantErr        bool
 	}{
 		{
+			// We receive an event from entity "foo", already in the system.
+			//
+			// We expect to retrieve the EntityConfig and EntityState for "foo"
+			// and the event to look like it came from "foo".
 			name:  "entity exists",
 			event: corev2.FixtureEvent("foo", "check-cpu"),
-			storeFunc: func(store *mockstore.MockStore) {
-				store.On("GetEntityByName", mock.Anything, "foo").
-					Return(corev2.FixtureEntity("foo"), nil)
+			storeFunc: func(s *storetest.Store, _ *corev2.Event) {
+				state := corev3.FixtureEntityState("foo")
+				config := corev3.FixtureEntityConfig("foo")
+
+				stateReq := storev2.NewResourceRequestFromResource(context.Background(), state)
+				configReq := storev2.NewResourceRequestFromResource(context.Background(), config)
+
+				wConfig, err := wrap.Resource(config)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				wState, err := wrap.Resource(state)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				s.On("Get", stateReq).Return(wState, nil)
+				s.On("Get", configReq).Return(wConfig, nil)
 			},
 			wantEntityName: "foo",
 		},
 		{
-			name:  "entity does not exist",
-			event: corev2.FixtureEvent("foo", "check-cpu"),
-			storeFunc: func(store *mockstore.MockStore) {
-				store.On("GetEntityByName", mock.Anything, "foo").
-					Return(nilEntity, nil)
-				store.On("UpdateEntity", mock.Anything, mock.AnythingOfType("*v2.Entity")).
-					Return(nil)
+			// We receive an event from entity "foo", currently unknown in the
+			// system.
+			//
+			// We expect an EntityConfig and an EntityState to be created for
+			// "foo" and the event to be mutated appropriately.
+			name: "entity does not exist",
+			event: &corev2.Event{
+				Check: corev2.FixtureCheck("check-cpu"),
+				Entity: &corev2.Entity{
+					ObjectMeta: corev2.ObjectMeta{
+						Name:      "foo",
+						Namespace: "default",
+					},
+					Subscriptions: []string{"linux"},
+				},
+			},
+			storeFunc: func(s *storetest.Store, e *corev2.Event) {
+				_, state := corev3.V2EntityToV3(e.Entity)
+				stateReq := storev2.NewResourceRequestFromResource(context.Background(), state)
+
+				config := corev3.FixtureEntityConfig("foo")
+				configReq := storev2.NewResourceRequestFromResource(context.Background(), config)
+
+				wState, err := wrap.Resource(state)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				wConfig, err := wrap.Resource(config)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				s.On("Get", mock.AnythingOfType("v2.ResourceRequest")).
+					Return(nilWrapper, &store.ErrNotFound{})
+
+				// Assert that CreateOrUpdate() was called with the expected
+				// request and wrapper
+				s.On("CreateOrUpdate", stateReq,
+					mock.MatchedBy(func(w *wrap.Wrapper) bool {
+						return w.String() == wState.String()
+					})).Return(nil)
+
+				// Assert that CreateIfNotExists() was called with the expected
+				// request and wrapper type
+				// TODO(ccressent): can we do something more strict with the
```

Comment:
Have you abandoned the TODO? I think it's fine as-is.

Reply:
I'm sure we can find a solution, but I'm punting on this...
Before, we would accept `mock.Anything` as argument. Now we only accept `*wrap.Wrapper` with the expected type information. The grail would be check the actual wrapped value on top of that. 

I've mostly abandoned it because other than type information, in a `Wrapper`, we only have a protobuf encoded (by default) `[]byte` to inspect and if I recall correctly, I would get 2 such slices that were not 100% exactly the same, even if they would still decode to the same object, and the test would fail on that ground.